rig/machine_control/machine_controller.py

```diff
@@ -326,14 +355,17 @@ def boot(self, width, height,
 
         # Actually boot the machine
         boot_kwargs.setdefault("boot_port", self.boot_port)
-        self.structs = boot.boot(self.initial_host, width=width, height=height,
-                                 **boot_kwargs)
+        self.structs = boot.boot(self.initial_host, **boot_kwargs)
         assert len(self.structs) > 0
 
-        # Check the machine is now booted
+        # Wait for the machine to completely boot
         if check_booted:
             try:
-                self.get_software_version(0, 0, 0)
+                p2p_address = (255, 255)
+                while p2p_address == (255, 255):
+                    time.sleep(0.1)
+                    p2p_address = self.get_software_version(
+                        255, 255, 0).position
```

Comment:
Should be set by the property when it is None. (The function which decides what SCP connection to use only uses _root_chip if it has been set since sending the command would require looking up the SCP connection to use which would [....])